File: src/main/java/de/sean/splugin/spigot/events/DeathEvent.java
package de.sean.splugin.spigot.events;

import de.sean.splugin.SPlugin;
import de.sean.splugin.util.SMessages;
import org.bukkit.ChatColor;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.PlayerDeathEvent;

public class DeathEvent implements Listener {
    @EventHandler
    public void PlayerDeath(PlayerDeathEvent event) {
        // Some more formatting in the future?
        final String message = SMessages.getRandomMessage("messages.death").replace("[message]", event.getDeathMessage()).replace("[player]", event.getEntity().getDisplayName());
        event.setDeathMessage(ChatColor.RED + message);
        
        /* Discord */
        SPlugin.discord.sendMessage(message);
    }
}

File: src/main/kotlin/de/sean/splugin/spigot/events/DeathEvent.kt
package de.sean.splugin.spigot.events

import de.sean.splugin.SPlugin
import de.sean.splugin.util.SMessages.getRandomMessage
import org.bukkit.ChatColor
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.PlayerDeathEvent

class DeathEvent : Listener {
    @EventHandler
    fun playerDeath(event: PlayerDeathEvent) {
        // Some more formatting in the future?
        val message = getRandomMessage("messages.death").replace("[message]", event.deathMessage!!).replace("[player]", event.entity.displayName)
        event.deathMessage = ChatColor.RED.toString() + message

        /* Discord */SPlugin.discord.sendMessage(message)
    }
}
